internal/pkg/template/templates/workloads/partials/cf/imported-alb-resources.yml

```diff
@@ -0,0 +1,142 @@
+# For one listener or the port 443 listener if two.
+{{- $arn := ""}}
+{{- if not $.ImportedALB.HasTwoListeners }}
+  {{- range $listener := $.ImportedALB.Listeners}}
+  {{- $arn = $listener.ARN}}
+  {{- end}}
+{{- else }}
+  {{- range $listener := $.ImportedALB.Listeners}}
+  {{- if eq $listener.Port 443}}
+  {{- $arn = $listener.ARN}}
+  {{- end}}
+  {{- end}}
+{{- end}}
+
+{{- if ne $arn ""}}
+HTTPRulePriorityAction:
+  Metadata:
+    'aws:copilot:description': 'A custom resource assigning priority for HTTP listener rules'
+  Type: Custom::RulePriorityFunction
+  Properties:
+    ServiceToken: !GetAtt RulePriorityFunction.Arn
+    RulePath: {{ fmtSlice (quoteSlice .ALBListener.RulePaths) }}
+    ListenerArn: {{$arn}}
+
+{{- $isHTTPS := .ALBListener.IsHTTPS}}
```

Comment:
I have a concern with using this condition for imported cert - 

It seems like `.ALBListener.IsHTTPS` is true in the following scenarios today:
1.  when there is an app-level domain
2. when the alb has an imported cert

With imported alb now, consider this scenario:
1. The user imports an ALB that has two listeners, one of them is an HTTPS listener.
2. The user does not have app-level domain nor imported cert.
3. They have a manifest that looks like this:
```yaml
http:
  alias:
    - name: my.example.com
       hosted_zone: hosted_zone_for_domain
  alb: my-alb-with-https-listener
```
Assuming the HTTPS listener already have a certificate that covers my.example.com.

Because [2] they have neither app-level domain nor imported cert, we won't create the listener rules on the HTTPS listener. This means that their HTTPS listener won't have a rule that route based on the alias, i.e. among other conditions, it'd miss ⬇️ . The users won't be able to access https://my.example.com (probably they'd be able to access http://my.example.com though, I haven't finished reviewing). 
```
      - Field: 'host-header'
        HostHeaderConfig:
          Values: {{fmtSlice (quoteSlice $rule.Aliases)}}
```

However, I think it's reasonable for them to expect https://my.example.com to have worked, given that they a) imported an ALB with the appropriate listeners and appropriate cert b) specified the alias in manifest with the appropriate hosted zone.

Reply:
What do you think of this:
1. Have an `IsHTTPS` boolean field under `template.ImportedALB`
2. Set `IsHTTPS` to `true` if there is an HTTPS listener (by checking the protocol, instead of the port) under the ALB. I think absolutely most of the time 443 means HTTPS, and actually I am not sure if ALB even allows otherwise... but checking the protocol removes the assumption that 443 = HTTPS.